sdk/storage/azure-storage-common/CMakeLists.txt

```diff
@@ -2,26 +2,27 @@
 # SPDX-License-Identifier: MIT
 
 cmake_minimum_required (VERSION 3.13)
+project(azure-storage-common LANGUAGES CXX)
 
-set(AZURE_STORAGE_COMMON_HEADER
-    inc/azure/storage/common/access_conditions.hpp
-    inc/azure/storage/common/account_sas_builder.hpp
-    inc/azure/storage/common/concurrent_transfer.hpp
-    inc/azure/storage/common/constants.hpp
-    inc/azure/storage/common/crypt.hpp
-    inc/azure/storage/common/file_io.hpp
-    inc/azure/storage/common/reliable_stream.hpp
-    inc/azure/storage/common/shared_key_policy.hpp
-    inc/azure/storage/common/storage_common.hpp
-    inc/azure/storage/common/storage_credential.hpp
-    inc/azure/storage/common/storage_exception.hpp
-    inc/azure/storage/common/storage_per_retry_policy.hpp
-    inc/azure/storage/common/storage_retry_policy.hpp
-    inc/azure/storage/common/version.hpp
-    inc/azure/storage/common/xml_wrapper.hpp
-)
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../../../cmake-modules")
+
+include(az_version)
+include(CodeCoverage)
+include(DefineTransportAdapter)
+include(doxygen_common)
+include(global_compile_options)
+
+set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
+set(THREADS_PREFER_PTHREAD_FLAG TRUE)
```

Comment:
Does `azure-core`  depend on `Threads` already? If so, we don't need to repeat it here.

Reply:
I think if I remove it from here, it would apply when building Core, but not when building this library. Let me to move it to global_compile_options.

Reply:
but `azure-storage-common` depends on `azure-core`, it will get these compile options from there.

Reply:
If you import via vcpkg, and not via building by loading the entire Azure SDK project and building it, it is possible. Plus, unless you use `PARENT_SCOPE` when setting cmake variables, they won't be seen by siblings or predecessors.

Reply:
OK, i'm not familiar with how vcpkg works.